utils/evaluation.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import print_function, division
+import os
+
+import time
+import argparse
+from glob import glob
+
+# from tensorflow import keras
+import numpy as np
+from keras.optimizers import Nadam
+
+from quality_metrics import psnr, uiq, sam, sre
+
+from data_utils import get_logger
+from patches import recompose_images, OpenDataFilesTest, OpenDataFiles
+from DSen2Net import s2model
+logger = get_logger(__name__)
+
+SCALE = 2000
+lr = 1e-4
+MODEL_PATH = "../models/"
+
+
+def write_final_dict(metric, metric_dict):
+    # Create a directory to save the text file of including evaluation values.
+    predict_path = "val_predict/"
+    if not os.path.exists(predict_path):
+        os.makedirs(predict_path)
+
+    with open(os.path.join(predict_path, metric + '.txt'), 'w') as f:
+        f.writelines('{}:{}\n'.format(k, v) for k, v in metric_dict.items())
+
+
+def predict_downsampled_img(path, folder, dset, border, final_name):
+
+    logger.info("Loading weight ...")
+    if args.run_60:
+        input_shape = ((4, None, None), (6, None, None), (2, None, None))  # type: ignore
+    else:
+        input_shape = ((4, None, None), (6, None, None))  # type: ignore
+        # create model
+    model = s2model(input_shape, num_layers=6, feature_size=128)
+    print("Symbolic Model Created.")
+
+    nadam = Nadam(
+        lr=lr, beta_1=0.9, beta_2=0.999, epsilon=1e-8, schedule_decay=0.004
+    )
+
+    model.compile(
+        optimizer=nadam, loss="mean_absolute_error", metrics=["mean_squared_error"]
+    )
+    print("Model compiled.")
+    model.count_params()
+
+    # model = keras.models.load_model(MODEL_PATH +"s2_030_lr_1e-05.hdf5")
+    model.load_weights(MODEL_PATH +"s2_032_lr_1e-04.hdf5")
```

Comment:
I guess we can replace this whole part with `keras.models.load_model` Would be good also if the path to the model was an argument so we can change that easily without touching code

Reply:
Yes but for now I could only use this.

Reply:
somehow with keras.models.load_model was failing, but it was the old weight, with the new one I have not tried.